modules/thermal_hydraulics/test/tests/components/heat_source_from_total_power/phy.conservation.i

```diff
@@ -90,8 +90,11 @@ energy_change = ${fparse power_fraction * power * t}
   type = Transient
   scheme = 'bdf2'
 
-  solve_type = 'PJFNK'
+  solve_type = 'NEWTON'
   line_search = 'basic'
+  petsc_options_iname = '-pc_type'
+  petsc_options_value = ' lu'
```

Comment:
ILU seemed to struggle with this test in parallel, so I added this and it helped a lot...